tests/js/multiPortConnector-tests.js

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright 2022, Tony Atkins
+ * Copyright 2011-2020, Colin Clark
+ *
+ *  Licensed under the MIT license, see LICENSE for details.
+ */
+/*global jqUnit */
+(function (fluid) {
+    "use strict";
+    var youme = fluid.registerNamespace("youme");
+
+    fluid.registerNamespace("youme.test.multiPortConnector");
+
+    // The `youme.system` component needs to complete its own setup for ports to be visible elsewhere.  Since this is
+    // the only sequence of asynchronous events we need to occur before we can run our tests, we handle it using a
+    // single listener rather than full Fluid IoC tests.
+    fluid.defaults("youme.test.multiPortConnector.testRunner", {
+        runTests: fluid.notImplemented,
+        components: {
+            midiSystem: {
+                options: {
+                    listeners: {
+                        "onReady.runTests": {
```

Comment:
Another benefit of the "creation is readiness" pattern is that you could use the onCreate promisified event as the return value from promisified test fixtures